File: initials-project-ios/Utility/View/InitialsViewConfigurator.m
//
//  InitialsViewConfigurator.m
//  initials-project-ios
//
//  Created by Vladislav Kleschenko on 5/12/18.
//  Copyright © 2018 Uladzislau Kleshchanka. All rights reserved.
//
#import <UIKit/UIKit.h>
#import "InitialsViewConfigurator.h"
#import "IPFont.h"

#define SEGMENT_COUNT 20.0

@implementation InitialsViewConfigurator

+ (void)configurateView:(UIView *)view width:(CGFloat)width withConfiguration:(IPCreationConfiguration *)configuration initials:(NSArray<NSString *> *)initials{
    NSMutableArray *constraints = [NSMutableArray array];

    CGFloat segment = width/SEGMENT_COUNT;
    
    
    for (NSUInteger i = 0; i < initials.count; i++) {
        UILabel *letterLabel = [[UILabel alloc] init];
        
        Pattern *pattern = (Pattern *) configuration.options[CreationOptionsPattern];
        IPColor *backgroundColor = (IPColor *) configuration.options[CreationOptionsBackgroundColor];
        IPColor *fontColor = (IPColor *) configuration.options[CreationOptionsFontColor];
        IPFont *fontFamily = (IPFont *) configuration.options[CreationOptionsFontFamily];
        CGFloat letterSize = pattern.letterPatterns[i].size;

        letterLabel.text = initials[i];
        letterLabel.font = [UIFont fontWithName:fontFamily.fontName size:letterSize];
        letterLabel.textColor = fontColor.color;
        letterLabel.translatesAutoresizingMaskIntoConstraints = NO;
        
        [view addSubview:letterLabel];
        view.backgroundColor = backgroundColor.color;
        
        [constraints addObject:[letterLabel.centerXAnchor constraintEqualToAnchor:view.centerXAnchor constant:(segment * (pattern.letterPatterns[i].x))]];
        [constraints addObject:[letterLabel.centerYAnchor constraintEqualToAnchor:view.centerYAnchor constant:(segment * (pattern.letterPatterns[i].y))]];
    }
    [NSLayoutConstraint activateConstraints:constraints];
}

@end

File: initials-project-ios/Utility/View/InitialsViewConfigurator.swift
//
//  InitialsViewConfigurator.swift
//  initials-project-ios
//
//  Created by Uladzislau Kleshchanka on 6/15/18.
//  Copyright © 2018 Uladzislau Kleshchanka. All rights reserved.
//

import UIKit

@objc class InitialsViewConfigurator: NSObject {
    
    private static let SEGMENT_COUNT: Float = 20.0
    
    class func configurateView(_ view: UIView, width: Float, withConfiguration configuration: IPCreationConfiguration, initials: [String]) {
        var constraints = [NSLayoutConstraint]()
        
        let segment = width/SEGMENT_COUNT
        
        for i in 0..<initials.count {
            let letterLabel = UILabel()
            
            let pattern = configuration.options[0] as! Pattern
            let backgroundColor = configuration.options[1] as! IPColor
            let fontColor = configuration.options[2] as! IPColor
            let fontFamily = configuration.options[3] as! IPFont
            let letterSize = pattern.letterPatterns[i].size
            
            letterLabel.text = initials[i]
            letterLabel.font = UIFont.init(name: fontFamily.fontName, size: CGFloat(letterSize))
            letterLabel.textColor = fontColor.color
            letterLabel.translatesAutoresizingMaskIntoConstraints = false
            
            view.addSubview(letterLabel)
            view.backgroundColor = backgroundColor.color
            
            constraints.append(letterLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor, constant: (CGFloat(segment * Float(pattern.letterPatterns[i].x)))))
            
            constraints.append(letterLabel.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: (CGFloat(segment * Float(pattern.letterPatterns[i].y)))))
        }
        
        NSLayoutConstraint.activate(constraints)
    }
}
